### PR TITLE
feat(testnet): `StaticTestnet` data dir persist mode

### LINKED ISSUE #185: Add configurable data dir to config for pubky-testnet
It's not possible to choose the data dir for `pubky-testnet`, it always uses an ephemeral tmp dir. It would be useful to point it at a persistent location for some local testing. 

That is, we should be able to:
- Specify a file for the the testnet to store its data
- Write its data contents to file (rather than the existing forced `StorageConfigToml::InMemory`)
- Stop the testnet and restart with the same path, with the same data persisting

### PR DESCRIPTION
**Pull request by 86667**
`StaticTestnet` uses an ephemeral temp directory, so all homeserver state (user data, files, config) is lost on shutdown. There is no way to persist a testnet environment, users have been using docker-compose presumably with production mainline - [discussion in issue](https://github.com/pubky/pubky-core/issues/185#issuecomment-3138343305). 

This is annoying for users and a gap which needs to be closed. Weve seen it brought up in telegram a couple times. 

An added bonus is that i've wanted to improve `StaticTestnet` for a while so used this opportunity to sneak in a refactor to make it easy to expand. I also improved docs to make it clearer how `EphemeralTestnet` and `StaticTestnet` differ and what they should each be used for.

Note that guides will be coming soon for how `pubky-testnet` can be used for development.  In the readme I intentionally stick to discussing only the crate's behaviour.

Here we:
- feat: Add a `persist <data_dir>` subcommand to the pubky-testnet binary allowing it to store homeserver state on disk across restarts
- feat: Seed a deterministic keypair on first run so the homeserver identity is stable and matches the in-memory default
- refactor: Introduce `StaticTestnetBuilder` with `.persistent(path)` and `.homeserver_config(path)` for programmatic use
- refactor: Other cleanup on `StaticTestnet` and `EphemeralTestnet`

### CODE REVIEW

File: pubky-testnet/src/static_testnet.rs [hunk 1]
```diff
@@ -2,73 +2,223 @@ use crate::pubky::Pubky;
...
+            }
+            StorageMode::Persistent(data_dir) => {
+                testnet
+                    .run_persistent_homeserver(data_dir, self.homeserver_config.as_deref())
+                    .await
+                    .map_err(|e| anyhow::anyhow!("Failed to run persistent homeserver: {}", e))?;
+                true
+            }
+        };
+        testnet.persistent = persistent;
```
Comment (MCarlomagno) [inline pubky-testnet/src/static_testnet.rs]: I feel `persistent` for a bool var is a bit confusing, I would rather use `is_persistent`

File: pubky-testnet/src/static_testnet.rs [hunk 2]
```diff
@@ -175,41 +326,426 @@ impl StaticTestnet {
...
 
         let homeserver = HomeserverApp::start_with_mock_data_dir(mock).await?;
         self.testnet.homeservers.push(homeserver);
         Ok(())
     }
 }
+
+/// Copy the source config file into the persistent data directory.
+/// Errors if a `config.toml` already exists at the destination.
+fn seed_config(source: &Path, persistent_dir: &PersistentDataDir) -> anyhow::Result<()> {
```
Comment (MCarlomagno) [inline pubky-testnet/src/static_testnet.rs]: nit: I usually try to clean up standalone functions by making them part of higher level structs/enums, in this case I would make it  a private function of `StaticTestnet`, but this is personal taste

File: pubky-testnet/src/static_testnet.rs [hunk 3]
```diff
@@ -175,41 +326,426 @@ impl StaticTestnet {
...
+    let config_path = persistent_dir.get_config_file_path();
+    if config_path.exists() {
+        anyhow::bail!(
+            "config.toml already exists at {}. Remove --homeserver-config to use the existing config, \
+             or delete the file to replace it.",
+            config_path.display()
+        );
+    }
+    std::fs::create_dir_all(persistent_dir.path())?;
+    std::fs::copy(source, &config_path)?;
```
Comment (MCarlomagno) [inline pubky-testnet/src/static_testnet.rs]: nit: also something I usually avoid is to leave full import chains inline like this, I prefer something like `fs::copy(source, &config_path)` instead, but very personal too

**Comment by MCarlomagno** [comment]
other than the nits, the README tells users to omit `?pubky-test=true` in persistent mode, but nothing enforces it. Then the persistent mode could route into the epheremeral path and create a fresh `pubky_test_{uuid}` database each run

File: pubky-testnet/src/static_testnet.rs (hunk 1 above)
Comment (86667) [reply pubky-testnet/src/static_testnet.rs]: Changed to `is_persistent`

**Comment by 86667** [comment]
> the README tells users to omit `?pubky-test=true` in persistent mode, but nothing enforces it.

@MCarlomagno good point. Ive added a check for the flag when in persistent mode

**Review by MCarlomagno — DISMISSED** [review]
Thanks @86667, there is still an edge case 

what if `config.toml` itself contains `database_url = "...?pubky-test=true"`? I think the check should run against the final `config.general.database_url`

I mark the PR as approved since this is not a functional blocker

**Comment by 86667** [comment]
@MCarlomagno good catch again - moved check later so that it applies to all configuration routes

**Review by MCarlomagno — APPROVED** [review]
